### DISCUSSION TITLE
Postgres 16 > 17 Required?

### DISCUSSION THREAD

**Discussion by ekalp** [teslamate-org/teslamate#4326] [General]
It seems that this project jumps each time a new version of Postgres releases. Is moving from 16 > 17 truly required, or is there support in the project for v16 until a certain time period?

**Comment by cwanja** [comment]
I was running v15 just fine until 1.31.0.

**Reply by ekalp** [reply]
Nice, still wondering what the supported range is for various versions of Postgres as it isn't a simple move to each new version and can cause quite a number of issues for folks on lower powered devices or those may not have the skills (see the other discussions when's searching Postgres).

**Comment by brianmay** [comment]
I think it is a matter of: There is nothing known that will mean the old versions will break functionality. But we can't test all of the the old versions, and as a result there could be something that slips through that we don't know about that causes breakage. So we recommend the latest version because it is easier to trouble shoot problems if everyone is on the same version.

**Reply by ekalp** [reply]
Makes sense. Still seems interesting to force a DB server version change each time a new one drops. It is fairly common that a project will support N-1 or N-2 versions for DB storage unless a breaking change occurs.

**Reply by cwanja** [reply]
No one forced it and in our [release notes](https://github.com/teslamate-org/teslamate/releases/tag/v1.31.0), it says we support all the way back to v12. At the moment, we support N-5. 
> Regarding PostgreSQL 17: TeslaMate uses PostgreSQL as database, this is an external dependency and needs to be updated by yourself. Although TeslaMate currently runs fine with PostgreSQL 12+ we strongly recommend upgrading to the latest supported version. We recommend that you do this as follows

**Reply by ekalp** [reply]
> No one forced it and in our [release notes](https://github.com/teslamate-org/teslamate/releases/tag/v1.31.0), it says we support all the way back to v12. At the moment, we support N-5.
> 
> > Regarding PostgreSQL 17: TeslaMate uses PostgreSQL as database, this is an external dependency and needs to be updated by yourself. Although TeslaMate currently runs fine with PostgreSQL 12+ we strongly recommend upgrading to the latest supported version. We recommend that you do this as follows

I read the latest release notes and missed the prior ones. 🤦‍♂️ Thanks!

**Reply by cwanja** [reply]
> > No one forced it and in our [release notes](https://github.com/teslamate-org/teslamate/releases/tag/v1.31.0), it says we support all the way back to v12. At the moment, we support N-5.
> 
> > 
> 
> > > Regarding PostgreSQL 17: TeslaMate uses PostgreSQL as database, this is an external dependency and needs to be updated by yourself. Although TeslaMate currently runs fine with PostgreSQL 12+ we strongly recommend upgrading to the latest supported version. We recommend that you do this as follows
> 
> 
> 
> I read the latest release notes and missed the prior ones. 🤦‍♂️ Thanks!

All good, probably should have carried the PostgreSQL notice to the smaller incremental update.
